tests/ops/op_math/test_parametrized_hamiltonian.py

```diff
@@ -0,0 +1,225 @@
+        pH = ParametrizedHamiltonian([1.2, f1, 2.3, f2], [qml.PauliX(i) for i in range(4)])
+        params = [4.5, 6.7]
+        t = 2
+        op = pH(params, t)
+
+        assert isinstance(op, qml.ops.Sum)
+        assert len(op) == 2
+
+        H_fixed = op[0]
+        H_parametrized = op[1]
+        expected_H_fixed = qml.op_sum(
+            qml.s_prod(1.2, qml.PauliX(0)), qml.s_prod(2.3, qml.PauliX(2))
+        )
+        expected_H_parametrized = qml.op_sum(
+            qml.s_prod(f1(params, t), qml.PauliX(1)), qml.s_prod(f2(params, t), qml.PauliX(3))
+        )
+
+        assert qml.equal(H_fixed, expected_H_fixed)
+        assert qml.equal(H_parametrized, expected_H_parametrized)
+
+
+class TestInteractionWithOperators:
+    """Test that arithmetic operations involving or creating ParametrizedHamiltonians behave as expected"""
+
+    ops = (
+        (qml.Hamiltonian([2], [qml.PauliZ(0)]), qml.s_prod(2, qml.PauliZ(0))),
+        (qml.ops.SProd(2, qml.PauliZ(0)), qml.s_prod(2, qml.PauliZ(0))),
+        (qml.PauliX(2), qml.s_prod(1, qml.PauliX(2))),
+        (qml.PauliX(2) @ qml.PauliX(3), qml.s_prod(1, qml.PauliX(2) @ qml.PauliX(3))),
+    )
+
+    @pytest.mark.parametrize("H, res", ops)
+    def test_add_other_operators(self, H, res):
+        """Test that a Hamiltonian, SProd, Tensor or Observable can be added to a
+        ParametrizedHamiltonian, and will be incorporated in the H_fixed term"""
+        pH = ParametrizedHamiltonian([f1, f2], [qml.PauliX(0), qml.PauliY(1)])
+        new_pH = pH + H
+
+        assert pH.H_fixed is None
+        assert qml.equal(new_pH.H_fixed, res)
+
+    def test_adding_two_parametrized_hamiltonians(self):
+        """Test that two ParametrizedHamiltonians can be added together and
+        the H_fixed and H_parametrized terms are correctly combined."""
+        pH1 = ParametrizedHamiltonian([1.2, f2], [qml.PauliX(0), qml.PauliY(1)])
+        pH2 = ParametrizedHamiltonian([2.3, f1], [qml.Hadamard(2), qml.PauliZ(3)])
+
+        # H_fixed now contains the fixed terms from both pH1 and pH2
+        new_pH = pH1 + pH2
+        assert qml.equal(new_pH.H_fixed[0], pH1.H_fixed)
+        assert qml.equal(new_pH.H_fixed[1], pH2.H_fixed)
+
+        # H_parametrized now contained the parametrized terms from both pH1 and pH2
+        parametric_term = new_pH.H_parametrized([1.2, 2.3], 0.5)
+        assert qml.equal(parametric_term[0], pH1.H_parametrized([1.2, 2.3], 0.5))
+        assert qml.equal(parametric_term[1], pH2.H_parametrized([1.2, 2.3], 0.5))
+
+    def test_fn_times_observable_creates_parametrized_hamiltonian(self):
+        """Test a ParametrizedHamiltonian can be created by multiplying a
+        function and an Observable"""
+        pH = f1 * qml.PauliX(0)
+        assert isinstance(pH, ParametrizedHamiltonian)
+        assert len(pH.H_fixed_coeffs) == 0
+        assert isinstance(pH.H_parametrized(param, 0.5), qml.ops.SProd)
```

Comment:
could you add one test with a more compllicated signature that `pH(param, t)` creates indeed the correct value (i.e. that the function is called in the correct expected way)

Reply:
How complicated is more complicated? Are you thinking more terms? Or some other element of complexity?

Reply:
I think this is still missing. I would say we need a test function for an example that includes params for `pwc` functions. Although maybe that should added in the `pwc` PR. Are there other more complicated examples we should be sure to test for?

Reply:
Yes I think the pwc should be separate. One example where args are passed as a list, one as a tuple, and one as an array, trying to achieve high variety, should cover most cases we want covered here!